test/e2e/operator_test.go

```diff
@@ -77,47 +82,87 @@ func newIngressController(name, ns, domain string, epType operatorv1.EndpointPub
 	}
 }
 
-func TestOperatorAvailable(t *testing.T) {
-	cl, _, err := getClient()
-	if err != nil {
-		t.Fatal(err)
+func clusterOperatorConditionMap(conditions ...configv1.ClusterOperatorStatusCondition) map[string]string {
+	conds := map[string]string{}
+	for _, cond := range conditions {
+		conds[string(cond.Type)] = string(cond.Status)
+	}
+	return conds
+}
+
+func operatorConditionMap(conditions ...operatorv1.OperatorCondition) map[string]string {
+	conds := map[string]string{}
+	for _, cond := range conditions {
+		conds[string(cond.Type)] = string(cond.Status)
 	}
+	return conds
+}
+
+func conditionsMatchExpected(expected, actual map[string]string) bool {
```

Comment:
Not even sure if this is necessary/desired — should we fail if some condition type we don't expect appears?

Reply:
Can address in a followup once we decide